src/test/v1.2/basic/fdc3.joinChannel.ts

```diff
@@ -26,41 +30,42 @@ export default () =>
       await fdc3.leaveCurrentChannel();
     });
 
-    it("(BasicJC1) Can join channel and broadcast", async () => {
-      const wrapper = wrapPromise();
+    it("(BasicJC1) Can join channel", async () => {
       const channels = await fdc3.getSystemChannels();
 
-      if (channels.length > 0) {
-        try {
-          await fdc3.joinChannel(channels[0].id);
-
-          const currentChannel = await fdc3.getCurrentChannel();
-
-          expect(currentChannel).to.not.be.null;
-
-          const gotContext = (c) => {
-            return true;
-          };
-
-          fdc3.addContextListener("someContext", (ctx) => {
-            if (ctx.type == "someContext") {
-              wrapper.resolve();
-            } else {
-              wrapper.reject("wrong context type");
-            }
-          });
-
-          currentChannel.broadcast({
-            type: "someContext",
-            id: { name: "hello" },
-          });
-
-          await wrapper.promise;
-        } catch (ex) {
-          assert.fail("Error while joining channel: " + (ex.message ?? ex));
-        }
-      } else {
-        assert.fail("No system channels available");
+      await fdc3.joinChannel(channels[0].id);
+
+      const current = await fdc3.getCurrentChannel();
+
+      expect(current).to.not.be.null;
+    });
+
+    it("(BasicJC2) Can join the correct system channel", async () => {
+      const [channel] = await fdc3.getSystemChannels();
+
+      await fdc3.joinChannel(channel.id);
+
+      const current = await fdc3.getCurrentChannel();
+
+      expect(current).to.eql(channel);
+    });
+
+    it("(BasicJC3) Receive NoChannelFound error when joining a non-existing channel", async () => {
+      const errorWrapper = wrapPromise();
+
+      try {
+        await fdc3.joinChannel("invalid-channel-name");
+        errorWrapper.reject("Should have rejected - there's no such channel");
+      } catch (error) {
```

Comment:
If OpenError.AppTimeout isn't testable due to insufficient documentation, then so is this. It can't be added to the tests .

Reply:
In fact *most* implementations of 1.2 allow you to join AppChannels - it was only in 2.0 that we started to clarify the intention that it should only be user channels that you join (and documented the relevant error). Hence, I definitely think it needs testing in 2.0 - but not here.